pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -212,7 +212,39 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 	lastTaskRun := &v1beta1.TaskRun{}
 
 	// for existing TaskRuns update the BuildRun Status, if there is no TaskRun, then create one
-	if err = r.client.Get(ctx, types.NamespacedName{Name: request.Name, Namespace: request.Namespace}, lastTaskRun); err != nil && apierrors.IsNotFound(err) {
+	if err = r.client.Get(ctx, types.NamespacedName{Name: request.Name, Namespace: request.Namespace}, lastTaskRun); err != nil {
+		if apierrors.IsNotFound(err) {
+			generatedTaskRun, err := r.createTaskRun(ctx, build, buildRun)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+
+			ctxlog.Info(ctx, "creating TaskRun from BuildRun", namespace, request.Namespace, name, generatedTaskRun.Name, "BuildRun", buildRun.Name)
+			if err = r.client.Create(ctx, generatedTaskRun); err != nil {
+				updateErr := r.updateBuildRunErrorStatus(ctx, buildRun, err.Error())
+				return reconcile.Result{}, handleError("Failed to create TaskRun if no TaskRun for that BuildRun exists", err, updateErr)
+			}
+		} else {
+			return reconcile.Result{}, err
+		}
+	} else if lastTaskRun.GetLabels() == nil || lastTaskRun.GetLabels()[buildv1alpha1.LabelBuildRunUid] == "" {
+		// there is a taskrun with the name that we want to pick that was not created by us
+		ctxlog.Info(ctx, "A TaskRun with the BuildRun's name already exists but is not created by this controller.", namespace, request.Namespace, name, request.Name)
```

Comment:
I am thinking should we also delete this taskrun although its `lastTaskRun.GetLabels() == nil || lastTaskRun.GetLabels()[buildv1alpha1.LabelBuildRunUid] == ""` and create a new one?

Reply:
For this condition to happen, somebody must have created a TaskRun through means other than a BuildRun because we always set this label. I am reluctant to delete it then, because that TaskRun is then not anymore an implementation detail of a BuildRun and is to be managed outside the scope of this project.